src/node/building_blocks.ml

```diff
@@ -33,6 +33,12 @@ let is_current_producer state ~key_hash =
   Some (current_producer.address = key_hash)
 let minimum_signable_time_between_epochs = 10.0
 let maximum_signable_time_between_epochs = 20.0
+
+(** Used to add a delay between a tezos operation being confirmed,
+  needs to be bigger than the polling interval for operations *)
+let minimum_waiting_period_for_tezos_operation = 5.0
```

Comment:
Is there also a maximum? Like before considering this operation is expired?

Reply:
In one sense Tezos operations never "expire" - so long as they are included in the canonical chain. However, this raises an interesting point: what if two nodes are following different forks of Tezos temporarily?

If only a minority is following a particular fork, their blocks won't be signed but the chain will continue, no problem.

If the majority is following a particular fork that turns out to be wrong (i.e, there is a re-org), then (@EduardoRFS correct me if I'm wrong) I think what will happen is when the next block is posted to Tezos (i.e, one based on the wrong fork), Deku will also fork in a sense. There will be two histories: the old history based on the wrong fork, and a new history based on the new fork. Eventually a block producer will produce a block following the new, correct history. Assuming everyone is following the correct history (and if they're using a Tezos node that's functioning properly, they will), they will eventually all agree. When a block from this new history gets posted to Tezos, it will represent the validators collectively overwriting the history. There will be no sequence of blocks to get from the block referencing the bad fork to the block referencing the good fork, but it's OK because our proof-of-authority consensus currently allows them to rewrite history.

Reply:
Thanks for the detailed answer! I appreciate, it helps me to better understand the blockchain 👶